programs/canonical-swap/src/lib.rs

```diff
@@ -173,6 +173,33 @@ pub mod canonical_swap {
 
         Ok(())
     }
+
+    /// Set mint authority to the original account that transferred
+    /// it to the PDA and close the canonical_data account
+    pub fn return_canonical_token_mint_authority(
+        ctx: Context<ReturnCanonicalTokenMintAuthority>,
+        canonical_mint_authority_bump: u8,
+    ) -> ProgramResult {
+        // Take over mint authority for canonical token
+        let cpi_accounts = SetAuthority {
+            current_authority: ctx.accounts.canonical_mint_authority.to_account_info(),
+            account_or_mint: ctx.accounts.canonical_mint.to_account_info(),
+        };
+
+        let authority_seeds = &[
+            CANONICAL_MINT_AUTHORITY_PDA_SEED,
+            &[canonical_mint_authority_bump],
+        ];
+
+        let cpi_ctx = CpiContext::new(ctx.accounts.token_program.to_account_info(), cpi_accounts);
+
+        token::set_authority(
+            cpi_ctx.with_signer(&[authority_seeds]),
+            AuthorityType::MintTokens,
+            Some(ctx.accounts.canonical_data.initializer),
+        )?;
```

Comment:
Yeah, I'm trying to find the best way to make this less dev-ruggable and not brick the balances of wrapped tokens if something goes wrong.

It would be a bit of a painful process to clean up all the wrapped tokens. You'd need to issue n ix's with n token accounts and then finalize with the return of authority.

Reply:
Yea mainly worried about the non rugged wrapped token case, e.g. if wormhole V3 isn't rugged but V2 is, we're effectively burning all the ERC20 RLY locked in V3

But yea it's probably too dev-ruggable if we just transfer all those to initializer in the non compromised case, so maybe just letting them get burnt is better

Maybe a way to selectively remove WrappedData from being swappable (e.g. an is_active flag) is a good compromise/relatively cheap to implement? This also gives us a way to soft pause the CanonicalData without closing it/transferring the authority by basically deactivating all the WrappedDatas